### PR TITLE
Feature 934 release stage doc

### LINKED ISSUE #934: Add definitions to the Release Guide for the stages of the release cycle
## Describe the Task ##
Add definitions to the Release Guide for the stages of the release cycle.
Here is a link to the [Release Retrospective](https://docs.google.com/document/d/1nYeeNtv114_mWAtncZhKSfunMCZ3c6hY9naBtP3nwB0/edit#) where Keith has some guidance.

### Time Estimate ###
<1 day of work

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
METplus User's Telecon (June 14, 2021)

### Funding Source ###
2792541 and 2702691

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [x] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by jprestop**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>

Read through the added documentation for typos and good formatting:
https://metplus.readthedocs.io/en/feature_934_release_stage_doc/Users_Guide/overview.html
https://metplus.readthedocs.io/en/feature_934_release_stage_doc/Release_Guide/index.html#releasecyclestages

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Review the documentation (links above) for typos, formatting, and content to ensure they sufficiently describe the stages of our release cycle.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[None needed]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **[11/09/21]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Review by georgemccabe — CHANGES_REQUESTED** [review]
The content looks good and I don't see any typos. I would suggest using ^ instead of bolding the sub-sub-sections under Development Release sub-section so that those section headers will generate links.

```
RST
# with overline, for parts
* with overline, for chapters
=, for sections
-, for subsections
^, for subsubsections
“, for paragraphs
```

**Comment by jprestop** [comment]
@georgemccabe Thanks for the suggestion!  I changed "Beta" and "Release Candidate (rc)" from being bolded to being subsubsections.

**Review by georgemccabe — APPROVED** [review]
Looks good to me!

**Comment by ksearight** [comment]
Looks good!  My only thought it if it would be useful to mention in the Release Guide how the `main_vX.Y` branch gets created.  The text `1. Official Release (e.g. vX.Y.Z) from the develop branch` doesn't mention that the `main_vX.Y` branch gets created when doing the Official Release.

**Comment by jprestop** [comment]
@ksearight Thanks for the feedback.  The Release Guide instructions for an official release for each component describe the process of creating the main_vX.Y branch, however, this summary doesn't specify that detail.  Do you think it would be helpful to add the part in bold below to the summary information?

> 1. Official Release (e.g. vX.Y.Z) from the develop branch **(becomes the new main_vX.Y branch)**
> 2. Bugfix Release (e.g. vX.Y.Z) from the corresponding main_vX.Y branch
> 3. Development Release (e.g. vX.Y.Z-betaN or vX.Y.Z-rcN) from the develop branch

File: docs/Release_Guide/index.rst [hunk 1]
```diff
@@ -2,7 +2,57 @@
...
+
+A bugfix release introduces no new features, but fixes bugs in previous
+official releases and targets the most critical bugs affecting users. 
+
+Instructions Summary
+====================
+
 Instructions are provided for three types of software releases:
 
 #. **Official Release** (e.g. vX.Y.Z) from the develop branch
```
Comment (ksearight) [inline docs/Release_Guide/index.rst]: Official Release (e.g. vX.Y.Z) from the develop branch (becomes the new main_vX.Y branch)

**Review by ksearight — COMMENTED** [review]
I suggest adding the extra text for the Official Release you mentioned.

File: docs/Release_Guide/index.rst (hunk 1 above)
Comment (jprestop) [reply docs/Release_Guide/index.rst]: Thanks @ksearight.  I made the suggested change.

**Review by ksearight — APPROVED** [review]
